### ISSUE TITLE
nginx requires webroot-setup.service provided by httpd

### ISSUE THREAD

**Issue by mrkz** [clearlinux/distribution#1146]
installing `nginx` bundle alone will not produce a functional http server as the nginx service contains
```
[Unit]
...
After=webroot-setup.service
...
```

The proper systemd unit (`/usr/lib/systemd/system/webroot-setup.service`) is provided by the `httpd-data` package (means it requires `httpd` at bundle level atm).

We should make `webroot-setup.service` server agnostic as it really runs a oneshot configuration setup, regardless of the web server (currently it relies heavily on having `httpd` installed:
```
% cat /usr/lib/systemd/system/webroot-setup.service
[Unit]
Description=Initialize the HTTP web document root (/var/www/html)
After=syslog.target
ConditionPathExists=!/var/www/html

[Service]
Type=oneshot
ExecStartPre=/usr/bin/mkdir -p /var/www/html
ExecStartPre=/usr/bin/cp -r /usr/share/httpd/www/icons /usr/share/httpd/www/error /usr/share/httpd/www/cgi-bin /var/www/
ExecStart=/usr/bin/cp /usr/share/httpd/www/html/index.html /var/www/html/

[Install]
WantedBy=multi-user.target
```

**Comment by bwarden** [comment]
We could move webroot-setup.service to a new package, webroot, and add it to Requires for both httpd and nginx.

**Comment by bwarden** [comment]
Wait... nginx package includes nginx-setup.service that does the same thing (create/populate /var/www/html). Is anything actually failing?

**Comment by mrkz** [comment]
@bwarden  yes, specifically the nginx service fails because it's asking for `After=webroot-setup.service`, maybe just changing this `After` would do.

**Comment by qzheng527** [comment]
Per my understanding, After= is a "loose coupling", and a service with such a statement would still run if the one named in the After= line isn't started/existed at all. So I think we are fine here.

**Comment by ketonik** [comment]
@mkrz

I can't offer direct solution to this issue. Before building a LEMP stack I did significant research.

I contacted Nginx for their clarifications on the branches they recommend nginx-mainline

Direct Quote direct from Nginx"
"We generally recommend using the mainline branch. This is where we commit all new features, performance improvements, and enhancements. We actively test and QA the mainline branch, and as the source of NGINX Plus builds it is completely suitable for production use."

NGINX Plus FYI is the commercially paid for version, and thus they say nginx-mainline open source is the branch used in Nginx Plus.

Now if you look at the release version of nginx and nginx-mainline bundles and packages for Clear Linux, it should be obvious that Nginx-mainline branch is Intel's preferred branch, since the nginx bundle is a legacy version.

Other indicators, nginx-mainline received HTTP/2, years before nginx branch. In fact current mainline 1.17.x branch has HTTP/3 planned.

**Comment by mrkz** [comment]
> Per my understanding, After= is a "loose coupling", and a service with such a statement would still run if the one named in the After= line isn't started/existed at all. So I think we are fine here.

You are right, I tried to reproduce again today in a clean environment without success.

> Wait... nginx package includes nginx-setup.service that does the same thing (create/populate /var/www/html). Is anything actually failing?

Actually, why does nginx has this reference to `httpd` setup anyway?
@ahkok  mind to shed some light here?

Perhaps we could remove this `After=webroot-setup.service` if not needed?

**Comment by bwarden** [comment]
> Actually, why does nginx has this reference to `httpd` setup anyway?

Probably just to prevent a race between the two.

**Comment by ahkok** [comment]
the webroot-setup service needs to become its own package, so that both bundles can include it. This is a fallout of the removal of the web-server-basic bundle I think
